### PR TITLE
Add flag for environment vars substitution in config file

### PR DESCRIPTION
**Pull request by Ashmita152**
Setting default value to false to keep backward compatibility

Signed-off-by: Ashmita Bohara <ashmita.bohara152@gmail.com>

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/tempo/issues/788

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: cmd/tempo/main.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 
 	"github.com/cortexproject/cortex/pkg/util/flagext"
 	"github.com/cortexproject/cortex/pkg/util/log"
+	"github.com/drone/envsubst"
```
Comment (annanay25) [inline cmd/tempo/main.go:24]: nit: These need to be shuffled around. PTAL at - https://github.com/grafana/tempo/blob/main/CONTRIBUTING.md#go-imports

**Review by annanay25 — COMMENTED** [review]
Thanks for working on this one @Ashmita152! 🚀  I've left a few comments.

File: cmd/tempo/main.go (hunk 1 above)
Comment (Ashmita152) [reply cmd/tempo/main.go]: I have updated it but I am still not sure if I placed it properly. Also I don't see the guidelines of CONTRIBUTING.md being followed everywhere: eg. https://github.com/grafana/tempo/blob/main/cmd/tempo-cli/main.go

File: cmd/tempo/main.go [hunk 2]
```diff
@@ -134,6 +136,14 @@ func loadConfig() (*app.Config, error) {
 			return nil, fmt.Errorf("failed to read configFile %s: %w", configFile, err)
 		}
 
+		if *expandEnvVars {
```
[the remark this answers is not in the document]
Comment (Ashmita152) [reply cmd/tempo/main.go]: I wasn't able to think of any other way so I did in the same way as config.file. I think we are doing it in such a way to give less priority to flags defined in config file than the ones passed via command line args. Is my understanding correct ?

[the remark this answers is not in the document]
Comment (annanay25) [reply cmd/tempo/main.go]: That's right.

So I retried with the changes you made and while my environment variables are getting correctly parsed & replaced, the defaults are not getting picked up if the env var is _not defined_: 

```
ingester:
  trace_idle_period: ${var:8s}               # the length of time after a trace has not received spans to consider it complete and flush it
```

^ this is not working for me as mentioned in the docs https://github.com/grafana/tempo/pull/796/files#diff-43f71a30c23dbefbf46d18377ad4db42b3298c81679a75cc0b6cccf783da71d1R37 

Could you confirm if its working for you?

File: cmd/tempo/main.go (hunk 1 above)
Comment (annanay25) [reply cmd/tempo/main.go]: Yup, you are correct, we need a step in `make fmt`. Feel free to pick that up if you can :)

File: cmd/tempo/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ashmita152) [reply cmd/tempo/main.go]: Nice catch. You're right, there is definitely some bug with the default value not passing properly. I will look into it.

[the remark this answers is not in the document]
Comment (Ashmita152) [reply cmd/tempo/main.go]: I think it should be `${var:-8s}`(from: https://github.com/drone/envsubst/blob/master/readme.md)

I just tested with storage.local.path and storage.wal.path and this works as expected. Updated the doc with it.

**Review by annanay25 — APPROVED** [review]
LGTM. Thanks @Ashmita152!

**Review by mapno — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**